common/src/main/kotlin/no/nav/su/se/bakover/common/CorrelationId.kt

```diff
@@ -1,6 +1,46 @@
 package no.nav.su.se.bakover.common
 
+import org.slf4j.MDC
+import java.util.UUID
+
+const val CorrelationIdHeader = "X-Correlation-ID"
+
 @JvmInline
 value class CorrelationId(val value: String) {
     override fun toString() = value
+
+    companion object {
+        fun generate(): CorrelationId = CorrelationId(UUID.randomUUID().toString())
+
+        /** Henter `X-Correlation-ID` fra MDC */
+        fun getCorrelationIdFromThreadLocal(): CorrelationId? {
```

Comment:
Hadde det vært diggere for konsumenter om de slapp å forholde seg til om denne eksisterte eller ikke? Er det unødvendig at konsumentene må velge om de vet at en id eksisterer (med tilhørende bangs), eller om de skal generere en ny? Kunne vi like gjerne bare eksponert en funksjon som håndterer `getOrCreate` og la alle kalle denne? Så lenge denne kaller videre til funksjonen som trigger stacktrace hvis det mangler vil vi vel klare å luke ut alt til slutt?

Reply:
Sure